src/sealock.c

```diff
@@ -1,51 +1,63 @@
 
-#include <stdlib.h>
 
-#include "load_time_averaged.h"
 #include "sealock.h"
+#include "load_time_averaged.h"
+#include "timestamp.h"
+
+#include <stdlib.h>
 
 int sealock_load_timeseries(sealock_state_t *lock, char *filepath) {
   int status = SEALOCK_OK;
+  double *time_column = NULL;
+  size_t num_rows = 0;
+
   // read csv data
   if (!load_time_averaged_timeseries(&lock->timeseries_data, filepath)) {
     lock->current_row = 0;
-    lock->times_len = get_csv_num_rows(&lock->timeseries_data);
-    if (lock->times_len == 0) {
-      return SEALOCK_ERROR;
-    }
-    lock->times = malloc(lock->times_len * sizeof(double));
-    if (lock->times != NULL) {
-      return get_csv_column_data(&lock->timeseries_data, "time", lock->times, lock->times_len);
-    } else {
-      lock->times_len = 0;
+    num_rows = get_csv_num_rows(&lock->timeseries_data);
+    if (!num_rows)
       return SEALOCK_ERROR;
+    time_column = malloc(num_rows * sizeof(double));
+    if (time_column != NULL) {
+      status = get_csv_column_data(&lock->timeseries_data, "time", time_column, num_rows);
+      if (status == SEALOCK_OK) {
+        lock->times = timestamp_array_to_times(time_column, num_rows);
+        if (lock->times) {
+          lock->times_len = num_rows;
+        } else {
+          status = SEALOCK_ERROR;
+        }
+      }
+      free(time_column);
+      return status;
     }
   }
+
   return SEALOCK_ERROR;
 }
 
-void _sealock_update_current_row(sealock_state_t *lock, double time) {
+void _sealock_update_current_row(sealock_state_t *lock, time_t time) {
```

Comment:
I see this now, should these be static? I don't think that the name should start with an underscore, same reason as before

Reply:
Good idea. It's what I actually meant to signal with the underscores anyway.